### ISSUE TITLE
RF05 raises IndexError on an all-underscore identifier, and false-positives on other dialects

### ISSUE THREAD

**Issue by devYRPauli** [sqlfluff/sqlfluff#8423]
### Search before asking

I searched the open issues and found nothing covering this. #6733 is a different RF05 false positive, on TSQL table-valued method aliases.

### What Happened

`RF05` raises `IndexError` on an identifier made only of underscores, and reports it as an internal error.

The rule strips underscores before it inspects the identifier:

```python
# We always allow underscores so strip them out
identifier = identifier.replace("_", "")
```

An identifier such as `_` therefore becomes an empty string, and the redshift temporary table check then reads index 0 of it.

### Expected Behaviour

`SELECT _ FROM t` lints cleanly. `_` holds no special characters, and the rule's own comment says underscores are always allowed.

### Observed Behaviour

On redshift, the rule crashes:

```
CRITICAL [RF05] Applying rule RF05 to 'q.sql' threw an Exception: string index out of range
  File "src/sqlfluff/rules/references/RF05.py", line 204, in _eval
    and identifier[0] == "#"
IndexError: string index out of range
L:   1 | P:   8 | RF05 | Unexpected exception: string index out of range;
                       | Could you open an issue at ...
```

On other dialects the `#` check is skipped, so the empty string reaches the alphanumeric test at line 221 and `"".isalnum()` is `False`, producing a false positive:

```
L:   1 | P:   8 | RF05 | Do not use special characters in identifiers.
```

I reproduced the false positive on postgres, snowflake, tsql and bigquery. On ansi a bare `_` does not parse, so the rule is never reached there.

### How to reproduce

`q.sql`:

```sql
SELECT _ FROM t
```

```
sqlfluff lint --dialect redshift --rules RF05 q.sql    # IndexError
sqlfluff lint --dialect postgres --rules RF05 q.sql    # false positive
```

### Dialect

redshift for the crash; postgres, snowflake, tsql and bigquery for the false positive.

### Version

sqlfluff 4.3.0, reproduced on `main`.
